src/Controller/FrameworksController.php

```diff
@@ -332,14 +336,26 @@ public function search(Request $request, int $page = 1)
         } catch (NotFoundException | PaginationException $e) {
             throw new NotFoundHttpException('Page not found', $e);
         }
-
-        $data = [
-            'query'         => $query,
-            'pagination'    => $results->getPagination(),
-            'results'       => $results,
-            'categories'    => FrameworkCategories::getAll(),
-            'pillars'       => FrameworkCategories::getAllPillars()
-        ];
+        if ($flag == 'guidedmatch'){
+            $data = [
+                'query'                      => $query,
+                'pagination'                 => $results->getPagination(),
+                'results'                    => $results,
+                'categories'                 => FrameworkCategories::getAll(),
+                'pillars'                    => FrameworkCategories::getAllPillars(),
+                'guided_match_flag'          => $flag,
+                'keywords'                   => ['linen', 'laptop', 'legal'],
```

Comment:
Thanks for this change. Let's please do the logic here in the controller rather than in the html. So only set guided_match_flag if the query matches the keywords. 
That keeps the html quite straightforward and only dependent on the flag.
Additionally when the list grows large, we wouldn't want to be sending it all to the templating engine.

Reply:
The flag is for the feature to show. It not part of the logic to show the area in normal circumstances so this is an unnecessary change. We talk about the fact that the matching would done in the controller and in the guided match API since there are on going talks. Till that has been implemented this is suitable for current acceptance criteria.